app/models/user/sign_in.rb

```diff
@@ -0,0 +1,47 @@
+# == Schema Information
+# Schema version: 20220225214524
+#
+# Table name: user_sign_ins
+#
+#  id         :bigint           not null, primary key
+#  user_id    :bigint
+#  ip         :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  country    :string
+#
+# Record medadata about User sign in activity.
+class User::SignIn < ApplicationRecord
+  default_scope { order(created_at: :desc) }
+
+  belongs_to :user, inverse_of: :sign_ins
+
+  before_create :create?
+
+  def self.purge
+    where('created_at < ?', retention_days.days.ago).destroy_all
+  end
+
+  def self.search(query)
+    joins(:user).references(:users).where(<<~SQL, query: query)
+      lower(user_sign_ins.ip) LIKE lower('%'||:query||'%') OR
+      lower(user_sign_ins.country) LIKE lower('%'||:query||'%') OR
+      lower(users.name) LIKE lower('%'||:query||'%') OR
+      lower(users.email) LIKE lower('%'||:query||'%')
+    SQL
+  end
+
+  def self.retention_days
+    AlaveteliConfiguration.user_sign_in_activity_retention_days
+  end
+
+  def other_users
+    self.class.where(ip: ip).where.not(user_id: user_id)
```

Comment:
Should return distinct User instances not User::SIgnIn instances.

```suggestion
    User.distinct.joins(:sign_ins).
      where(user_sign_ins: { ip: ip }).
      where.not(user_sign_ins: { user_id: user_id })
```

Reply:
Or:

```suggestion
    User.distinct.joins(:sign_ins).
      where(user_sign_ins: { ip: ip }).
      where.not(id: user_id)
```

Reply:
Also add a failing spec